mailbox/postgres/src/main/java/org/apache/james/mailbox/postgres/PostgresMailboxSessionMapperFactory.java

```diff
@@ -122,4 +124,11 @@ public AttachmentMapper createAttachmentMapper(MailboxSession session) {
     public AttachmentMapper getAttachmentMapper(MailboxSession session) {
         throw new NotImplementedException("not implemented");
     }
+
+    public DeleteMessageListener deleteMessageListener() {
```

Comment:
is this method just for supporting testing?

Reply:
Mark `@VisibleForTesting` then?

Reply:
Little refactoring to have this protected (solves the issue then and no need of the tag right?) and using this method as well in the PostgresMailboxManagerProvider

Good now? :)